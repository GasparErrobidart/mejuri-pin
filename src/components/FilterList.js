import styled              from 'styled-components'
import React               from 'react'
import {connect}           from 'react-redux'
import { FiCheck }         from 'react-icons/fi';

import { setFilter }       from '../actions/filterActions'

const FlexColumn = styled.div`
  margin: 40px 0 0 0 ;
  display: flex;
  flex-direction: column;
  justify-content: center;
  flex-wrap:wrap;
`

const FilterItemWrapper = styled.div`
  display: flex;
  padding: 10px;
  border:1px solid ${({theme})=> theme.colors.lightGray};
  background: white;

  &:hover,&:active{
    border-color: ${({theme})=> theme.colors.middleGray}
  }

  & *{
    transition: all 200ms ease-in;
    color : ${({checked,theme}) => checked ? theme.colors.darkGray : theme.colors.darkMiddleGray}
  }

  & div:first-child{
    overflow: hidden;
    flex: 0 0 ${({checked}) => checked ? "25px" : 0};
    height: 15px;
  }
`

const FilterItem = (props)=>{
  const {children,checked} = props
  let icon = checked ? <FiCheck/> : null
  return(
    <FilterItemWrapper {...props}>
      <div>
        {icon}
      </div>
      <div>
        {children}
      </div>
    </FilterItemWrapper>
  )
}


class FilterList extends React.Component{

  constructor(props){
    super(props)
  }

  changeFilter(filter){
    let filters = [...this.state.filters]
    filters.find()
  }

  handleClick(filter){
    console.log("Handling click",filter)
    this.props.setFilter(filter.id,!filter.checked)
  }

  render(){
    return(
      <FlexColumn>
        {
          this.props.filter.map(
              (item, i)=> (
                <FilterItem key={item.id} onClick={this.handleClick.bind(this,item)} checked={item.checked}>{item.word}</FilterItem>
            )
          )
        }
      </FlexColumn>
    )
  }

}


const mapStateToProps = (state) => {
  return {
      filter : state.filter
  };
};

const mapDispatchToProps = (dispatch) => {
    return {
        setFilter: (id,checked) => {
            dispatch(setFilter(id,checked));
        }
    };
};

export default connect(mapStateToProps,mapDispatchToProps)(FilterList);
