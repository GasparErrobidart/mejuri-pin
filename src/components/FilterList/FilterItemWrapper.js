import styled from 'styled-components'

const FilterItemWrapper = styled.div`
  display: flex;
  padding-left: 30px;
  &:hover,&:active{
    border-color: ${({theme})=> theme.colors.middleGray};
  }
  cursor:pointer;

  & *{
    transition: all 200ms ease-in;
    color : ${({checked,theme}) => checked ? theme.colors.darkGray : theme.colors.darkMiddleGray};
  }

  &:hover *{
    color : ${({checked,theme}) => theme.colors.darkGray };
  }

  & div:first-child{
    overflow: hidden;
    flex: 0 0 ${({checked}) => checked ? "25px" : 0};
    height: 15px;
  }
`

export default FilterItemWrapper
