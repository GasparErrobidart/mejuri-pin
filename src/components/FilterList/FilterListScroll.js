import styled              from 'styled-components'

const FilterListScroll = styled.div`
  overflow-y:scroll;
  max-height:50vh;
  border-top: 1px solid ${({theme})=> theme.colors.lightGray};
  @media (min-width:768px){
    max-height:auto;
    overflow-y:visible;
    border:none;
  }
`

export default FilterListScroll
