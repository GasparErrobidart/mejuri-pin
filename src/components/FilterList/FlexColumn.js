import styled              from 'styled-components'

const FlexColumn = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  flex-wrap:wrap;
  position:fixed;
  top:60px;
  left:0;
  width:100%;
  @media (min-width:768px){
    position:static;
    top:auto;
    left:auto;
    width:auto;
  }
`

export default FlexColumn
