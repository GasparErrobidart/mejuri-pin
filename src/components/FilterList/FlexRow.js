import styled              from 'styled-components'

const FlexRow = styled.div`
  display: flex;
  & > *:first-child{
    flex-grow:1;
  }
`

export default FlexRow
