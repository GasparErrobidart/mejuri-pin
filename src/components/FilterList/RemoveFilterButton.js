import styled              from 'styled-components'

const RemoveFilterButton = styled.div`
  color: ${({theme})=> theme.colors.darkMiddleGray};
  margin-left: auto;
  cursor:pointer;
`

export default RemoveFilterButton
