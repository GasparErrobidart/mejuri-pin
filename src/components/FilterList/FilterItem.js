import React                from 'react'
import StyledFilterItem     from './StyledFilterItem'
import FilterItemWrapper    from './FilterItemWrapper'
import { FiCheck }          from 'react-icons/fi';

const FilterItem = (props)=>{
  const {children,checked} = props
  let icon = checked ? <FiCheck/> : null
  return(
    <StyledFilterItem {...props}>
      <FilterItemWrapper {...props}>
          <div>
            {icon}
          </div>
          <div>
            {children}
          </div>
      </FilterItemWrapper>
    </StyledFilterItem>
  )
}

export default FilterItem
