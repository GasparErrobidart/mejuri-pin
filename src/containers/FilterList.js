// DEPENDENCIES
import styled               from 'styled-components'
import React                from 'react'
import {connect}            from 'react-redux'
import {
  FiChevronDown,
  FiChevronUp,
  FiX
}/*---------------------->*/from 'react-icons/fi';

// COMPONENTS
import Container            from '../components/Container'
import FlexColumn           from '../components/FilterList/FlexColumn'
import FlexRow              from '../components/FilterList/FlexRow'
import RemoveFilterButton   from '../components/FilterList/RemoveFilterButton'
import FilterItem           from '../components/FilterList/FilterItem'
import StyledFilterItem     from '../components/FilterList/StyledFilterItem'
// ACTIONS
import {
  addFilter,
  removeFilter
}/*---------------------->*/from '../actions/filterActions'





class FilterList extends React.Component{

  constructor(props){
    super(props)
    this.state = {
      open : typeof window == "object" && window.innerWidth >= 768
    }
    this.toggleList = this.toggleList.bind(this)
    this.clearFilters  = this.clearFilters.bind(this)
  }

  changeFilter(filter){
    let filters = [...this.state.filters]
    filters.find()
  }

  handleClick(filter){
    console.log("Handling click")
    if(!this.isChecked(filter)){
      this.props.addFilter(filter)
    }else{
      this.props.removeFilter(filter)
    }
  }

  toggleList(){
    this.setState(
      (prevState)=>{
        return {...prevState,open : !prevState.open}
      }
    )
  }

  clearFilters(){
    this.props.filter.forEach(
      filter =>{
        this.props.removeFilter(filter)
      }
    )
  }

  isChecked(category){
    return this.props.filter.find( filter => filter.id === category.id)
  }

  renderList(){
    let categories = Object.keys(this.props.categories).map( id => this.props.categories[id] )
    categories.unshift({
      id   : "liked",
      name : "Liked"
    })
    return categories.map(
        (item, i)=> (
          <FilterItem key={item.id} onClick={this.handleClick.bind(this,item)} checked={this.isChecked(item)}>{item.name}</FilterItem>
      )
    )
  }

  render(){
    let filterCount = this.props.filter.length
    let filterDisplay = (
      <StyledFilterItem>
        <Container>
          <FlexRow>
            <div onClick={this.toggleList}>
                Filter {filterCount > 0 ? `(${filterCount})` : null} { this.state.open ? <FiChevronUp/> : <FiChevronDown/> }
            </div>
            { filterCount > 0 && (
              <RemoveFilterButton onClick={this.clearFilters}>
                  <FiX/> Clear filter
              </RemoveFilterButton>
            )}
          </FlexRow>
        </Container>
      </StyledFilterItem>
    )
    return(
      <FlexColumn>
      {filterDisplay}
        { this.state.open &&
          this.renderList()
        }
      </FlexColumn>
    )
  }

}

const mapStateToProps = (state) => {
  return {
      filter : state.filter,
      likes : state.likes,
      categories : state.categories.data,
      products : state.products.data
  };
};

const mapDispatchToProps = (dispatch) => {
    return {
      addFilter: (filter) => {
          dispatch(addFilter(filter));
      },
      removeFilter: (filter) => {
          dispatch(removeFilter(filter));
      }
    };
};

export default connect(mapStateToProps,mapDispatchToProps)(FilterList);
