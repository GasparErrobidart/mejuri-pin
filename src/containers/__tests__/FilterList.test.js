import '@testing-library/jest-dom'
import 'jest-styled-components'

import configureStore    from 'redux-mock-store'
import React             from 'react'
import { cleanup }       from '@testing-library/react'
import { Provider }      from 'react-redux'
import renderer          from 'react-test-renderer'
import {ThemeProvider}   from 'styled-components'

import FilterList    from '../FilterList'
import theme             from '../../components/GlobalStyle/themes/mejuri-pin'

const mockStore = configureStore([]);


describe('FilterList component', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
        filter : [],
        likes : [],
        categories : { data :
          {
            "0" : { name : "test category" , id: "0"}
          }
        },
        products : { data : {} }
    });
  });

  it("Should render",()=>{
    const component       = renderer.create(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <FilterList/>
        </ThemeProvider>
      </Provider>
    );
    const tree            = component.toJSON()
    expect(tree).toMatchSnapshot();
  })

  afterEach(cleanup)


});
