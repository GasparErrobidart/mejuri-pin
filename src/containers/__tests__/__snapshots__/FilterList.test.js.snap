// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterList component Should render 1`] = `
.c2 {
  padding: 0 20px;
  box-sizing: border-box;
  width: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  position: fixed;
  z-index: 1;
  top: 60px;
  left: 0;
  width: 100%;
  -webkit-transition: opacity 300ms ease-in;
  transition: opacity 300ms ease-in;
  opacity: 0.2;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 > *:first-child {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c4 {
  overflow-y: scroll;
  max-height: 50vh;
  border-top: 1px solid #F3F3F3;
}

.c1 {
  background: white;
  padding: 15px 0;
  border-top: 1px solid #F3F3F3;
  cursor: pointer;
}

.c5 {
  background: white;
  padding: 15px 0;
  border-top: 1px solid #F3F3F3;
  cursor: default;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 30px;
  cursor: pointer;
}

.c6:hover,
.c6:active {
  border-color: #CCCCCC;
}

.c6 * {
  -webkit-transition: all 200ms ease-in;
  transition: all 200ms ease-in;
  color: #888888;
}

.c6:hover * {
  color: #303030;
}

.c6 div:first-child {
  overflow: hidden;
  -webkit-flex: 0 0 0;
  -ms-flex: 0 0 0;
  flex: 0 0 0;
  height: 15px;
}

@media (min-width:768px) {
  .c0 {
    position: static;
    top: auto;
    left: auto;
    width: auto;
  }
}

@media (min-width:768px) {
  .c4 {
    max-height: auto;
    overflow-y: visible;
    border: none;
  }
}

@media (min-width:768px) {
  .c1 {
    background: none;
  }
}

@media (min-width:768px) {
  .c5 {
    background: none;
  }
}

<div
  className="c0"
  disabled={true}
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        <div
          onClick={[Function]}
        >
          Filter 
           
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="18 15 12 9 6 15"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div
    className="c4"
  >
    <div
      className="c5"
      onClick={[Function]}
    >
      <div
        className="c6"
        onClick={[Function]}
      >
        <div />
        <div>
          Liked
        </div>
      </div>
    </div>
    <div
      className="c5"
      onClick={[Function]}
    >
      <div
        className="c6"
        onClick={[Function]}
      >
        <div />
        <div>
          test category
        </div>
      </div>
    </div>
  </div>
</div>
`;
